iroha/src/isi.rs

```diff
@@ -21,9 +21,14 @@ pub enum Instruction {
     Asset(crate::asset::isi::AssetInstruction),
     /// Variant of instructions related to `Account`.
     Account(crate::account::isi::AccountInstruction),
-    /// This variant of Iroha Special Instructions composes two other instructions into one, and
+    /// Variant of instructions related to `Permission`.
+    Permission(crate::permission::isi::PermissionInstruction),
+    /// This variant of Iroha Special Instruction composes two other instructions into one, and
     /// executes them both.
     Compose(Box<Instruction>, Box<Instruction>),
+    /// This variant of Iroha Special Instruction executes the first instruction and if it succeeded
+    /// executes the second one, if failed - the third one.
+    If(Box<Instruction>, Box<Instruction>, Box<Instruction>),
```

Comment:
Maybe make second and third instructions optional? Sometimes it makes sense to execute instruction only on success and also sometimes only on failure.

We can also imitate this behavior with `DoNothing` instruction.

Reply:
Will align set of instruction to needs of users